Jenkinsfile

```diff
@@ -25,45 +25,34 @@ pipeline {
     }    
 
     stages {
-        stage('Build and Test Locally') {
-            stages {
-                stage('Static Code Analysis') {
-                    steps {
-                        script {
-                            if (sh(label: 'Build docker image for static code analysis', script: 'docker build -t local/111-static-code-analysis:${BUILD_TAG} -f Dockerfile.tests .', returnStatus: true) != 0) {error("Failed to build docker image for static code analysis")}
-                            if (sh(label: 'Running docker container', script: 'docker run -it -d --name check-container-${BUILD_TAG} local/111-static-code-analysis:${BUILD_TAG} /bin/bash', returnStatus: true) != 0) {error("Failed to run docker container")}
-                            if (sh(label: 'Running static code analysis', returnStdout: true, script: 'docker exec check-container-${BUILD_TAG} /bin/bash -c "./gradlew staticCodeAnalysis --continue"', returnStatus: true) != 0) {error("Static code analysis failed.")}
-                        }
-                    }
-                    post {
-                        always {
-                            sh label: 'Create reports directory', script: 'mkdir -p build/reports'
-                            sh label: 'Copy reports to jenkins', script: 'docker cp check-container-${BUILD_TAG}:/home/gradle/service/build/reports/. ./build/reports'
-                            sh label: 'Stop docker container', script: 'docker stop check-container-${BUILD_TAG}'
-                            sh label: 'Remove docker container', script: 'docker rm check-container-${BUILD_TAG}'
-                            recordIssues(
-                                enabledForFailure: true,
-                                tools: [
-                                    checkStyle(pattern: 'build/reports/checkstyle/*.xml'),
-                                    spotBugs(pattern: 'build/reports/spotbugs/*.xml')
-                                ]
-                            )
-                        }
-                    }
+        stage('Test') {
+            steps {
+                script {
+                    if (sh(label: 'Build image for tests', script: 'docker build -t local/111-tests:${BUILD_TAG} -f Dockerfile.tests .', returnStatus: true) != 0) {error("Failed to build docker image for tests")}
+                    if (sh(label: 'Running docker tests container', script: 'docker run -v /var/run/docker.sock:/var/run/docker.sock --name 111-tests-${BUILD_TAG} local/111-tests:${BUILD_TAG} ./gradlew check -i --continue --rerun-tasks', returnStatus: true) != 0) {error("Failed to run docker tests container")}
```

Comment:
`gradle check` does all the testing

Reply:
The `-v /var/run/docker.sock:/var/run/docker.sock` gives image control over docker so int tests can spawn their own conntainers at runtime